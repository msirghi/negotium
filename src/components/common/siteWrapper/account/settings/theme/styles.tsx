import { makeStyles } from '@mui/styles';

export const useThemeSettingsStyles = makeStyles({
  title: {
    fontSize: 14,
    padding: '12px 0'
  },
});
