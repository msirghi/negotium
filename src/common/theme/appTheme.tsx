// @ts-nocheck
import { createTheme } from '@mui/material/styles';
import colors from '../styles/colors';

const defaultFontFamily = `"Open Sans", "Roboto", "Helvetica", "Arial", sans-serif`;

export const defaultLightTheme = createTheme({
  typography: {
    allVariants: {
      fontFamily: defaultFontFamily,
    },
  },
  palette: {
    primary: {
      main: colors.primaries.lightBlue,
      light: colors.primaries.lightBlue_1,
      dark: colors.primaries.lightBlue,
    },
    custom: {
      menuIconBackground: colors.primaries.lightBlue,
      headerSearchBackground: colors.blues.light,
    },
  },
});

export const noirAppTheme = createTheme({
  typography: {
    allVariants: {
      fontFamily: defaultFontFamily,
    },
  },
  palette: {
    primary: {
      main: colors.greys['900'],
      light: colors.greys['800'],
      dark: colors.primaries.teal,
    },
    custom: {
      menuIconBackground: colors.greys['800'],
      headerSearchBackground: colors.greys['800'],
    }
  },
});

export const themeMap: { [key: string]: any } = {
  default: defaultLightTheme,
  noir: noirAppTheme,
};
