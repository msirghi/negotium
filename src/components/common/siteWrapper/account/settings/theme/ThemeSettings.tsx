import { ThemeItem } from './item/ThemeItem';
import { siteThemes } from '../../../../../../common/constants/constants';
import { Grid } from '@mui/material';
import { useThemeSettingsStyles } from './styles';
import { useDispatch, useSelector } from 'react-redux';
import { RootState } from '../../../../../../redux/store';
import { setSiteTheme } from '../../../../../../redux/account/accountSlice';
import AccountService from '../../../../../../services/AccountService';

export const ThemeSettings = () => {
  const classes = useThemeSettingsStyles();
  const siteTheme = useSelector(
    (state: RootState) => state.account.metadata.theme
  );
  const dispatch = useDispatch();

  const onThemeChange = async (theme: string) => {
    if (siteTheme !== theme) {
      dispatch(setSiteTheme(theme));
      await AccountService.updateUserTheme(theme);
    }
  };

  return (
    <div>
      <div className={classes.title}>
        Choose a theme that matches your preference.
      </div>
      <Grid container spacing={3}>
        {siteThemes.map((theme) => (
          <Grid key={theme.color} item>
            <ThemeItem
              theme={theme}
              selected={siteTheme === theme.internalKey}
              onClick={onThemeChange}
            />
          </Grid>
        ))}
      </Grid>
    </div>
  );
};
