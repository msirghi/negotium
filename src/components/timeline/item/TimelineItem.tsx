import { Task } from '../../../common/types/tasks.types';
import { FC } from 'react';
import { useTimelineItemStyles } from './styles';
import DateUtils from '../../../common/utils/dateUtils';
import { useTheme } from '@mui/system';

type Props = {
  task: Task;
  onClick: (task: Task) => void;
  active: boolean;
};

export const TimelineItem: FC<Props> = ({ task, onClick, active }) => {
  const theme = useTheme();
  const defaultColor = theme.palette.primary.light;
  const activeColor = theme.palette.primary.main;
  const classes = useTimelineItemStyles({ active, activeColor, defaultColor });
  const date = DateUtils.formatDateForTask(task.dueDate);

  const handleClick = () => onClick(task);

  return (
    <div className={classes.container} onClick={handleClick}>
      <p className={classes.title}>{task.title}</p>
      {date.value && <p className={classes.date}>{date.value}</p>}
    </div>
  );
};
