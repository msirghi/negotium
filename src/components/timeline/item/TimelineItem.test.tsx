import { TasksMock } from '../../../common/tests/mockData/tasks-mock';
import { mount } from 'enzyme';
import { TimelineItem } from './TimelineItem';
import renderer from 'react-test-renderer';
import { Checkbox } from '@mui/material';
import { ThemeProvider } from '@mui/system';
import { noirAppTheme } from '../../../common/theme/appTheme';

describe('TimelineItem', () => {
  const defaultProps = {
    task: TasksMock[0],
    onClick: jest.fn(),
    active: false,
    markAsDone: jest.fn(),
  };

  const renderComponent = () => {
    return (
      <ThemeProvider theme={noirAppTheme}>
        <TimelineItem {...defaultProps} />
      </ThemeProvider>
    );
  };

  afterEach(jest.clearAllMocks);

  it('should match the snapshot', () => {
    const wrapper = renderer.create(renderComponent());
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle marking task as done', () => {
    const wrapper = mount(renderComponent());
    const checkbox = wrapper.find(Checkbox);

    // @ts-ignore
    checkbox.props().onChange();
    expect(defaultProps.markAsDone).toBeCalled();
  });

  it('should handle task click', () => {
    const wrapper = mount(renderComponent());
    const container = wrapper.find('div').at(0);

    container.simulate('click');
    expect(defaultProps.onClick).toBeCalled();
  });
});
