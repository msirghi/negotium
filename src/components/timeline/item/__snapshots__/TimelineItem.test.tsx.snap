// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItem should match the snapshot 1`] = `
<div
  className="makeStyles-container-1 makeStyles-container-5"
  onClick={[Function]}
>
  <div
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  >
    <span
      className="MuiCheckbox-root MuiCheckbox-colorWarning MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorWarning PrivateSwitchBase-root Mui-checked makeStyles-checkbox-4 css-1azr8hg-MuiButtonBase-root-MuiCheckbox-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <input
        checked={true}
        className="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="CheckBoxIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <div
      className="MuiBox-root css-0"
    >
      <p
        className="makeStyles-title-2"
      >
        [{"type":"paragraph","children":[{"text":"new task for 123 "},{"text":""}]}]
      </p>
      <p
        className="makeStyles-date-3"
      >
        12 Nov, 2021
      </p>
    </div>
  </div>
</div>
`;
